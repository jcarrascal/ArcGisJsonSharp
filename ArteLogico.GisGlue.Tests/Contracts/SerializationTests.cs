using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ArteLogico.GisGlue.Contracts;
using Newtonsoft.Json;
using NUnit.Framework;

namespace ArteLogico.GisGlue.Tests.Contracts
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("SerializationTestsApprovals")]
    public class SerializationTests
    {
        [Test]
        public void RoundTripSerialization_WhenGivenAWebMapWithAnOperationalLayer_ThenMatchesTheInput()
        {
            // Based on a sample given by Esri on http://resources.arcgis.com/en/help/arcgis-web-map-json/index.html#/ArcGIS_map_service_operational_layer/02qt00000018000000/
            this.CheckRoundTripSerialization<WebMap>("SerializationTestsApprovals");
        }

        [Test]
        public void RoundTripSerialization_WhenGivenAWebMapWithASingleBasemapLayer_ThenMatchesTheInput()
        {
            // Based on a sample given by Esri on http://resources.arcgis.com/en/help/arcgis-web-map-json/index.html#/Single_basemap_layer/02qt00000016000000/
            this.CheckRoundTripSerialization<WebMap>("SerializationTestsApprovals");
        }

        [Test]
        public void RoundTripSerialization_WhenGivenAWebMapWithASingleOperationalLayer_ThenMatchesTheInput()
        {
            // Based on a sample given by Esri on http://resources.arcgis.com/en/help/arcgis-web-map-json/index.html#/Single_layer_from_an_ArcGIS_map_service_as_an_operational_layer/02qt0000001m000000/
            this.CheckRoundTripSerialization<WebMap>("SerializationTestsApprovals");
        }

        [Test]
        public void RoundTripSerialization_WhenGivenAWebMapWithFeatureServiceOperationalLayers_ThenMatchesTheInput()
        {
            // Based on a sample given by Esri on http://resources.arcgis.com/en/help/arcgis-web-map-json/index.html#/ArcGIS_feature_service_operational_layers/02qt00000017000000/
            this.CheckRoundTripSerialization<WebMap>("SerializationTestsApprovals");
        }

        [Test]
        public void RoundTripSerialization_WhenGivenAWebMapWithTwoBasemapLayers_ThenMatchesTheInput()
        {
            // Based on a sample given by Esri on http://resources.arcgis.com/en/help/arcgis-web-map-json/index.html#/Two_basemap_layers/02qt00000015000000/
            this.CheckRoundTripSerialization<WebMap>("SerializationTestsApprovals");
        }
    }
}
