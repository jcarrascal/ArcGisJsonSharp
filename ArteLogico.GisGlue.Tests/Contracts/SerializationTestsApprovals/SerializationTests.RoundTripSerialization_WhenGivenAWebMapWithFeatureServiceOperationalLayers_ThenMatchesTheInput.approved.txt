{
  "baseMap": {
    "baseMapLayers": [
      {
        "id": "World_Imagery_2704",
        "opacity": 1,
        "url": "http://services.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer",
        "visibility": true
      },
      {
        "id": "World_Boundaries_and_Places_2640",
        "isReference": true,
        "opacity": 1,
        "url": "http://services.arcgisonline.com/ArcGIS/rest/services/Reference/World_Boundaries_and_Places/MapServer",
        "visibility": true
      }
    ],
    "title": "Imagery with Labels"
  },
  "operationalLayers": [
    {
      "id": "Geology_8280",
      "mode": 1,
      "opacity": 1.0,
      "popupInfo": {
        "description": null,
        "fieldInfos": [
          {
            "fieldName": "objectid",
            "isEditable": false,
            "label": "Objectid",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "emp_id",
            "isEditable": true,
            "label": "Employee Id",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "fault_type",
            "isEditable": true,
            "label": "Fault Type",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "fault_subtype",
            "isEditable": true,
            "label": "Fault Subtype",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": true
          },
          {
            "fieldName": "classification_confidence",
            "isEditable": true,
            "label": "Confidence In Classifcation",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": true
          },
          {
            "fieldName": "fault_basis",
            "isEditable": true,
            "label": "Fault Basis",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": true
          },
          {
            "fieldName": "shear_criteria",
            "isEditable": true,
            "label": "Shear Criteria",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": true
          },
          {
            "fieldName": "strike",
            "isEditable": true,
            "label": "Strike",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "dip",
            "isEditable": true,
            "label": "Dip",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "dipdirection",
            "isEditable": true,
            "label": "Dipdirection",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "plunge",
            "isEditable": true,
            "label": "Plunge",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "trend",
            "isEditable": true,
            "label": "Trend",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "rake",
            "isEditable": true,
            "label": "Rake",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "confidence_measure",
            "isEditable": true,
            "label": "Confidence Of Measurement",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": true
          },
          {
            "fieldName": "measurement_method",
            "isEditable": true,
            "label": "Measurement Method",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": true
          },
          {
            "fieldName": "comments",
            "isEditable": true,
            "label": "Comments",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": true
          },
          {
            "fieldName": "globalid",
            "isEditable": false,
            "label": "GlobalID",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          }
        ],
        "mediaInfos": [],
        "showAttachments": true,
        "title": "{fault_type}"
      },
      "title": "Faults",
      "url": "http://sampleserver5.arcgisonline.com/arcgis/rest/services/Energy/Geology/FeatureServer/6",
      "visibility": true
    },
    {
      "id": "Geology_3137",
      "mode": 1,
      "opacity": 1.0,
      "popupInfo": {
        "description": null,
        "fieldInfos": [
          {
            "fieldName": "objectid",
            "isEditable": false,
            "label": "Objectid",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "emp_id",
            "isEditable": true,
            "label": "Employee Id",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "fold_type",
            "isEditable": true,
            "label": "Fold Type",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "classification_confidence",
            "isEditable": true,
            "label": "Confidence In Classification",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": true
          },
          {
            "fieldName": "strike",
            "format": {
              "digitSeparator": true,
              "places": 0
            },
            "isEditable": true,
            "label": "Strike",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "dip",
            "format": {
              "digitSeparator": true,
              "places": 0
            },
            "isEditable": true,
            "label": "Dip",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "dipdirection",
            "format": {
              "digitSeparator": true,
              "places": 0
            },
            "isEditable": true,
            "label": "Cipcirection",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "plunge",
            "format": {
              "digitSeparator": true,
              "places": 0
            },
            "isEditable": true,
            "label": "Plunge",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "trend",
            "format": {
              "digitSeparator": true,
              "places": 0
            },
            "isEditable": true,
            "label": "Trend",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "rake",
            "format": {
              "digitSeparator": true,
              "places": 0
            },
            "isEditable": true,
            "label": "Rake",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          },
          {
            "fieldName": "confidence_measurement",
            "isEditable": true,
            "label": "Confidence In Measurement",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": true
          },
          {
            "fieldName": "measurement_method",
            "isEditable": true,
            "label": "Measurement Method",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": true
          },
          {
            "fieldName": "comments",
            "isEditable": true,
            "label": "Comments",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": true
          },
          {
            "fieldName": "fold_subtype",
            "isEditable": true,
            "label": "Fold Subtype",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": true
          },
          {
            "fieldName": "globalid",
            "isEditable": false,
            "label": "GlobalID",
            "stringFieldOption": "textbox",
            "tooltip": "",
            "visible": false
          }
        ],
        "mediaInfos": [],
        "showAttachments": true,
        "title": "{fold_type}"
      },
      "title": "Folds",
      "url": "http://sampleserver5.arcgisonline.com/arcgis/rest/services/Energy/Geology/FeatureServer/7",
      "visibility": true
    }
  ],
  "version": "1.6"
}